h5netcdf/core.py

```diff
@@ -121,6 +120,72 @@ def __getitem__(self, key):
         return self._h5ds[key]
 
     def __setitem__(self, key, value):
+        # This conditional is used to automatically resize the underlying HDF5
+        # data set if the corresponding dimension is unlimited and if the value
+        # requires it.
+        if self._h5ds.maxshape and None in self._h5ds.maxshape:
```

Comment:
Can you move this logic for inferring the new size for a variable into a separate helper function? That would make things a little easier to test.